src/main/kotlin/no/nav/syfo/application/SenOppfolgingService.kt

```diff
@@ -49,24 +49,24 @@ class SenOppfolgingService(
     }
 
     fun publishUnpublishedKandidatStatus(): List<Result<SenOppfolgingKandidat>> {
-        val unpublishedKandidater = senOppfolgingRepository.getUnpublishedKandidater()
-        val publishedKandidater = unpublishedKandidater.map { kandidat ->
-            kandidatStatusProducer.sendKandidat(kandidat = kandidat)
+        val publishedKandidater = senOppfolgingRepository.getUnpublishedKandidater().map { kandidat ->
+            kandidatStatusProducer.send(kandidat = kandidat)
                 .map {
                     senOppfolgingRepository.setKandidatPublished(it.uuid)
                     it
                 }
         }
-        val unpublishedVurderinger = senOppfolgingRepository.getUnpublishedVurderinger()
-        val publishedVurderinger = unpublishedVurderinger.map { (kandidat, vurdering) ->
-            kandidatStatusProducer.sendVurdering(vurdering = vurdering, kandidat = kandidat)
-                .map {
-                    senOppfolgingRepository.setVurderingPublished(vurdering.uuid)
-                    it
-                }
-        }
+        val publishedKandidaterWithVurderinger =
+            senOppfolgingRepository.getKandidaterWithUnpublishedVurderinger().map { kandidat ->
+                kandidatStatusProducer.send(kandidat = kandidat)
+                    .map {
+                        val latestUnpublishedVurdering = kandidat.getLatestUnpublishedVurdering()!!
+                        senOppfolgingRepository.setVurderingPublished(latestUnpublishedVurdering.uuid)
+                        it
+                    }
+            }
```

Comment:
Må det være to prosesser? Kan ikke `getUnpublishedKandidater` hete `getUnpublishedKandidatStatuser` og så publiserer man bare basert på både `kandidat.published_at` og evt `vurdering.published_at`. Og så vil de statusene som ikke er en vurdering ha `sisteVurdering = null`.

Reply:
Hm, mulig man kan slå det sammen på et vis, men vi måte om vi skal oppdatere `publishedAt` på kandidaten eller vurderingen. Forslag? :)

Reply:
Prøvde meg på
```
SELECT kandidat.* 
                FROM SEN_OPPFOLGING_KANDIDAT kandidat 
                LEFT JOIN SEN_OPPFOLGING_VURDERING vurdering on vurdering.kandidat_id = kandidat.id 
                WHERE vurdering.published_at IS NULL OR kandidat.published_at IS NULL
                ORDER BY kandidat.created_at ASC
```
for å hente ut alt i samme spørring, men da blir kandidater uten vurdering hentet ut hver gang fordi `vurdering.published_at IS NULL`. Måtte i så fall sjekket på nytt i servicen at kandidat.publishedAt eller vurdering.publishedAt er null før vi publiserer :thinking: Begynner å bli litt hårete logikk der da, så kanskje bedre å la det være to spørringer og to operasjoner.

Reply:
Tror jeg fant en grei løsning nå :)